### PR TITLE
[Matrix] Enable joint_matrix_fill for joint_matrix feature

### PR DESCRIPTION
**Pull request by yubingex007-a11y**

### CODE REVIEW

File: sycl/test/matrix/matrix-int8-test-fill.cpp
Comment (dkhaldi) [inline sycl/test/matrix/matrix-int8-test-fill.cpp:71]: Instead of adding a new test, I would suggest changing the existing one to use fill instead of load for C.

File: sycl/test/matrix/matrix-int8-test-fill.cpp
Comment (dkhaldi) [inline sycl/test/matrix/matrix-int8-test-fill.cpp:72]: load for C is redundant here

File: sycl/include/CL/__spirv/spirv_ops.hpp
Comment (MrSidims) [inline sycl/include/CL/__spirv/spirv_ops.hpp:93]:
```suggestion
__spirv_CompositeConstruct(const T &v);
```
Note, there is not Scope flag for the instruction, but it should be OK, since it presence in the result type.

Comment (MrSidims) [reply sycl/include/CL/__spirv/spirv_ops.hpp:93]:
```suggestion
__spirv_JointMatrixFillINTEL(const T v, __spv::Scope::Flag Sc = S);
```
Like const T& the compiler will emit a temporary reference of value T, while we want to pass a constant expression to the instructions below and not load from this temporary reference.

**Comment by yubingex007-a11y** [comment]
If joint_matrix_fill(const T v)   call __spirv_CompositeConstruct( const Tv), just like https://godbolt.org/z/bxobascv8, in IR before spv, __spirv_CompositeConstruct will not have a immediate operand. Then, spir-v translator will feel difficult to lower it since llvm.experimental.matrix.fill has a immediate operand.

So, we suggest that we make joint_matrix_fill a macro function, then __spirv_CompositeConstruct  will carry a immediate value.

**Comment by yubingex007-a11y** [comment]
Ping?

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (MrSidims) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:198]: scope is unused here

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (MrSidims) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:200]:
```suggestion
joint_matrix_fill(joint_matrix<T, NumRows, NumCols, Layout, Group> &res,
                          const T v) {
```

`sg` was unused

Comment (dkhaldi) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:200]: Since we are not using scope on the SPIRV instruction as we are using the existing spirv_CompositeConstruct instruction. This argument will remain unused.
But we still need it on the DPC++ function to match the other DPC++ functions

Comment (MrSidims) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:200]: In this case please cast it to void, otherwise the compiler will emit a diagnostic.

**Review by MrSidims — COMMENTED** [review]
Other than unused scope the patch is LGTM.

**Review by dkhaldi — DISMISSED** [review]
LGTM

**Comment by yubingex007-a11y** [comment]
ping again.

**Review by MrSidims — APPROVED** [review]

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (MrSidims) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:201]: nit: might be worth to put a comment here, why we kept (sg) parameter (to be aligned with other AMX API?, to may be replaced __spirv_CompositeConstruct with some other instruction in case if we are not satisficed without breaking API/ABI with returning 'sg' back?)

**Review by dkhaldi — DISMISSED** [review]
LGTM

**Comment by yubingex007-a11y** [comment]
ping? @intel/llvm-reviewers-runtime

**Review by vladimirlaz — APPROVED** [review]

**Comment by yubingex007-a11y** [comment]
@bader is the fail in pre-checkin is known issue?

**Comment by bader** [comment]
> @bader is the fail in pre-checkin is known issue?

I haven't seen `DiscardEvents/invalid_event.cpp` failures before, but it's better to ask test owners.

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by yubingex007-a11y** [comment]
@intel/llvm-reviewers-runtime ping?
The basic_event_collection.cpp is expected I think. I saw https://github.com/intel/llvm-test-suite/pull/658 disable this case.

**Comment by yubingex007-a11y** [comment]
@bader ping?

**Comment by bader** [comment]
> @bader ping?

Waiting on code owner review from intel/llvm-reviewers-runtime.

> The basic_event_collection.cpp is expected I think. I saw [intel/llvm-test-suite#658](https://github.com/intel/llvm-test-suite/pull/658) disable this case.

The test was re-enabled by https://github.com/intel/llvm-test-suite/pull/659

**Comment by yubingex007-a11y** [comment]
@intel/llvm-reviewers-runtime ping?

**Comment by yubingex007-a11y** [comment]
@vladimirlaz ping?

**Comment by yubingex007-a11y** [comment]
The failed test in SYCL / Default Linux / HIP AMD GPU Test Suite can also be seen other PRs

**Comment by yubingex007-a11y** [comment]
@bader ping

**Comment by vladimirlaz** [comment]
@AGindinson is working on free_function_* failure @psamolysov-intel  is working on AtomicRef failures. The ticket is ready for merge

**Comment by bader** [comment]
> The ticket is ready for merge

@vladimirlaz, please, approve and merge.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**
